docs/src/userguide/language_basics.rst

```diff
@@ -381,6 +381,48 @@ Sometimes Cython will complain unnecessarily, and sometimes it will fail to
 detect a problem that exists. Ultimately, you need to understand the issue and
 be careful what you do.
 
+
+Type Casting
+------------
+
+As C uses the ``"("`` and ``")"`` for casting types, Cython uses ``"<"`` and ``">"``
+to signify a casting. For example::
+
+    cdef char *p
+    cdef float *q
+    p = <char*>q
+
+If one of the types is a Python object for ``<type>x``, Cython will try to do a coercion.
+
+ .. note:: Cython will not stop a casting where there is no conversion, but it will emit a warning.
```

Comment:
Maybe: "Cython will not prevent a redundant cast, but emits a warning for it." ? Is that what you meant here?